File: examples/java-ddd/application/src/main/java/org/myddd/java/application/assembler/UserAssembler.java
package org.myddd.java.application.assembler;

import org.myddd.java.api.dto.UserDTO;
import org.myddd.java.domain.User;

import java.util.Objects;

public class UserAssembler {

    public static UserDTO toUserDTO(User user){
        if(Objects.isNull(user))return null;
        var userDTO = new UserDTO();
        userDTO.setId(user.getId());
        userDTO.setUserId(user.getUserId());
        userDTO.setName(user.getName());
        userDTO.setEmail(user.getEmail());
        userDTO.setPhone(user.getPhone());
        return userDTO;
    }

    public static User toUser(UserDTO userDTO){
        if(Objects.isNull(userDTO))return null;
        var user = new User();
        if(Objects.nonNull(userDTO.getId()))user.setId(userDTO.getId());
        user.setUserId(userDTO.getUserId());
        user.setName(userDTO.getName());
        user.setEmail(userDTO.getEmail());
        user.setPhone(userDTO.getPhone());
        return user;
    }
}

File: samples/kotlin-ddd/application/src/main/kotlin/org/myddd/kotlin/application/assembler/UserAssembler.kt
package org.myddd.kotlin.application.assembler

import org.myddd.kotlin.api.dto.UserDTO
import org.myddd.kotlin.domain.User

fun User.toDTO():UserDTO {
    return UserDTO(id = id, userId = userId, name = name, phone = phone, email = email)
}

fun UserDTO.toUser():User {
    requireNotNull(userId)
    requireNotNull(name)

    val user = User()
    user.id = id
    user.userId = userId!!
    user.name = name!!
    user.phone = phone
    user.email = email
    return user
}